cvxpy/tests/test_attributes.py

```
@@ -81,6 +81,12 @@ def test_sparsity_assign_value(self):
         prob.solve()
         assert np.allclose(X.value, z)
         
+        z = np.zeros((3, 3))
+        z[A.sparse_idx] = [-1, -2, -3, -4]
+        A.value_sparse = sp.coo_array(([-1, -3, -2, -4], [(0, 1, 2, 2), (0, 2, 1, 2)]))
```

I think this should raise an error because the coordinates are in a different order and I don't want to commit to sorting them.

---

The data and coordinates are in a consistent order. Isn't the `coo_array` equivalent to

`sp.coo_array(([-1, -2, -3, -4], [(0, 2, 1, 2), (0, 1, 2, 2)]))`?

If assigning two equivalent sparse arrays (identical when converted to dense) to parameter `A` gives different solutions `X`, then there is a bug that we need to fix (and this test revealed the bug).

Maybe you had something else in mind - which behavior do you suggest? After a parameter is specified with sparsity `pattern`, any value assignment should follow the same order of coordinates?

---

> consistent order

I'm not certain what this means

> any value assignment should pattern without any re-ordering of coordinates

Correct. if we remove your change in this PR to the error, this line errors saying that the coordinates are invalid. We could give a better error message that makes it clear that the order is wrong, but I think we should raise an error.

Otherwise, we are committing to sorting the coordinates, and I don't think we should do that.